Update app header

yoshiemuranaka opened the pull request:
- Create WebSocket for host status in app header
- Visual style updates to move us closer to proposed UI
- Enabled https for devServer to allow secure WebSocket (wss)
- Sass loader config changes to allow sass variables to be
   used in single file components

yoshiemuranaka committed:
Add host status plugin

Adding host status plugin to create WebSocket and
get host state changes from server.

- Changed webpack devServer to https to allow for
  secure WebSocket creation (wss)
- Updates to AppHeader to visually indicate changes
  in host state
- Cleaned up api.js file

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Remove ws property on proxy server

Creates sockjs CORS errors, and is not needed
to proxy ws requests.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Check if user is logged in when creating WebSocket

Adds check if user is already authenticated so WebSocket
is created when browser refreshed.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Update GlobalStore module

Return 'Unreachable' state in UI when host transition
on state instead of 'Rebooting', since 'Rebooting' can be
misleading.
Including small parameter name changes.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Update header items

Clean up some header items to move towards proposed
UI update. Still needs a lot of styling modificatinos
but moving closer to agreed changes. Removed hostname/IP
and updated status items in navigation.

- Created a status icon component
- Added base _obmc-custom.scss file to vue config
  sass loader to make sass variables available in
  single file components

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka commented:
waiting for #51 to be merged so I can reference event log store for server health status in header

derick-montague reviewed (approved):
Looks good. Can't seem to run on localhost anymore with the websocket, but can run on the IP that the dev server is running on for anyone else trying to test and for future work once this is merged in.

gtmills commented on src/store/modules/GlobalStore.js:
```diff
@@ -1,10 +1,35 @@
+  error: 'xyz.openbmc_project.State.Host.HostState.Quiesced',
+  transition: {
+    on: 'xyz.openbmc_project.State.Host.Transition.On',
+    reboot: 'xyz.openbmc_project.State.Host.Transition.Reboot'
```
Also could have an off transition, beyond the scope of this.

gtmills commented on src/store/modules/GlobalStore.js:
```diff
@@ -1,10 +1,35 @@
+      return 'Off';
+    case HOST_STATE.error:
+      return 'Error';
+    case HOST_STATE.transition.reboot:
```
how would we ever hit this case? 
HostState can only ever be running, off, or Quiesced, so we hit those first...

https://github.com/openbmc/phosphor-dbus-interfaces/blob/master/xyz/openbmc_project/State/Host.interface.yaml#L35

yoshiemuranaka replied:
I added these transition states because they were being returned with the websocket events. I've seen transition.on and transition.reboot

yoshiemuranaka replied:
I'm checking for either CurrentHostState or RequestedHostTransition in the websocket event

gtmills commented on src/store/plugins/HostStatusPlugin.js:
```diff
@@ -0,0 +1,37 @@
+    interfaces: ['xyz.openbmc_project.State.Host']
+  };
+
+  const initWebSocket = () => {
```
A comment or 2 in here on *why* we are doing this could help when we upstream. Also maybe a link to the document that explains how all this works might also be helpful https://github.com/openbmc/docs/blob/b41aff0fabe137cdb0cfff584b5fe4a41c0c8e77/rest-api.md#event-subscription-protocol

yoshiemuranaka replied:
Great documentation..hadn't seen this earlier. I will plan to add comments

gtmills commented on src/store/modules/GlobalStore.js:
```diff
@@ -1,10 +1,35 @@
+const hostStateMapper = hostState => {
+  switch (hostState) {
+    case HOST_STATE.on:
+      return 'Running';
```
Why "Running"? but then we store it as "on"?

yoshiemuranaka replied:
Yeah, was thinking to change this. I was originally following the copy we have existing in our angular app, but with the new design proposals, it looks like the status will be indicated with the icon only and no text

gtmills commented:
Looks good! Thanks Yoshie! just had a couple of comments, if you fix up/answer, feel free to merge!

yoshiemuranaka committed:
Update host status mapper

Originally used similar copy to existing Angular app that showed
status 'Running' in the app header, since new designs no longer show
text to indicate host state, made changes to the returned host status
values.
Adding Off transition state to HOST_STATE constants object.
Also including some naming updates to statusIcon to make types more
general.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Add comments to WebSocketPlugin

Added some comments and changed plugin name from
HostStatusPlugin to WebSocketPlugin to be more generic.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

derick-montague commented:
Migrated to webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28729
